src/EditorFeatures/Core/Implementation/Suggestions/SuggestedActionsSource.cs

```diff
@@ -860,6 +890,48 @@ private void OnSuggestedActionsChanged(Workspace currentWorkspace, DocumentId cu
 
                 Volatile.Write(ref _lastSolutionVersionReported, solutionVersion);
             }
+
+            public async Task<ISuggestedActionCategorySet> GetSuggestedActionCategoriesAsync(ISuggestedActionCategorySet requestedActionCategories, SnapshotSpan range, CancellationToken cancellationToken)
+            {
+                var provider = _owner;
+                var result = new List<string>();
+                using (var asyncToken = _owner.OperationListener.BeginAsyncOperation(nameof(GetSuggestedActionCategoriesAsync)))
+                {
+                    var selection = await GetSpanAsync(range).ConfigureAwait(false);
+                    if (selection != null)
+                    {
+                        var document = range.Snapshot.GetOpenDocumentInCurrentContextWithChanges();
+                        if (await HasFixesAsync(provider, document, range, cancellationToken).ConfigureAwait(false))
+                        {
+                            result.Add(PredefinedSuggestedActionCategoryNames.CodeFix);
+                            result.Add(PredefinedSuggestedActionCategoryNames.ErrorFix);
```

Comment:
Correct me if I'm wrong, but it looks like as long as any 'fix' is present, we're returning 'ErrorFix' as one of the categories with applicable actions.

Based on the mappings in the IDE :
- ErrorFix is of higher precedence than CodeFix
- ErrorFix maps to the 'error' lightbulb icon
- CodeFix maps to the 'default' lightbulb icon

Therefore, if 1 or more fixes are applicable, the error lightbulb icon will always be displayed on mouse hover over squiggle, or on set caret position.

The thing to keep in mind is that the lightbulb API 1) calls GetSuggestedActionCategoriesAsync(), picks the category of highest precedence, and then displays the icon for that category. On lightbulb expansion with Ctrl+. however the lightbulb displays the icon for the highest precedence category name in the list of SuggestedActionsSets.